R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,10 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (nzchar(master)) {
+    assign("spark.master", master, envir = sparkConfigMap)
```

Comment:
I don't know why we prefer named parameter especially for `master`. Because in spark-submit, it seems we prefer `--master` over `--conf spark.master`.  And considering `master` is the first parameter, it gives users the impression that it is prefered. 

The following command will run on local mode
```
bin/spark-shell --master local --conf spark.master=yarn-client
```